From d2e1b968afe83be089d2009725e545491310f258 Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd@collabora.com>
Date: Sat, 22 May 2021 14:17:19 +0200
Subject: [PATCH] Let pipewire-pulse services conflict with their pulseaudio
 counterparts

You can only run either pulseaudio *or* the pipewire-pulse daemon at one
time. So make the pipewire-pulse units conflict with their pulseaudio
counterparts to avoid both running at the same time
---
 src/daemon/systemd/user/pipewire-pulse.service.in | 1 +
 src/daemon/systemd/user/pipewire-pulse.socket     | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/daemon/systemd/user/pipewire-pulse.service.in b/src/daemon/systemd/user/pipewire-pulse.service.in
index 061e2acb..56b8a64f 100644
--- a/src/daemon/systemd/user/pipewire-pulse.service.in
+++ b/src/daemon/systemd/user/pipewire-pulse.service.in
@@ -17,6 +17,7 @@ Requires=pipewire-pulse.socket
 ConditionUser=!root
 Wants=pipewire.service
 After=pipewire.service
+Conflicts=pulseaudio.service
 
 [Service]
 LockPersonality=yes
diff --git a/src/daemon/systemd/user/pipewire-pulse.socket b/src/daemon/systemd/user/pipewire-pulse.socket
index 39d2c897..1ae5edaf 100644
--- a/src/daemon/systemd/user/pipewire-pulse.socket
+++ b/src/daemon/systemd/user/pipewire-pulse.socket
@@ -1,6 +1,7 @@
 [Unit]
 Description=PipeWire PulseAudio
 ConditionUser=!root
+Conflicts=pulseaudio.socket
 
 [Socket]
 Priority=6
-- 
2.30.2

